File: 1288. Remove Covered Intervals/Solution.java
class Solution {
    public int removeCoveredIntervals(int[][] intervals) {
        
        TreeMap<Integer, Integer> map = new TreeMap<>();
        
        int remaining = intervals.length;
        
        for (int[] interval : intervals) {
            // check if the interval is covered by other intervals
            Map.Entry floorEntry = map.floorEntry(interval[0]);
            if (floorEntry != null && (int) floorEntry.getValue() >= interval[1]) {
                remaining--;
            } else {
                // check if the interval covers other intervals... 
                Map.Entry ceilingEntry = map.ceilingEntry(interval[0]);
                while (ceilingEntry != null && (int) ceilingEntry.getValue() <= interval[1]) {
                    map.remove(ceilingEntry.getKey());
                    remaining--;
                    ceilingEntry = map.ceilingEntry((int) ceilingEntry.getKey()+1);
                }
                map.put(interval[0], interval[1]);
            }
        }
        
        return remaining;
    }
}

File: 1288. Remove Covered Intervals/Solution.py
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        
        sorted_intervals = sorted(intervals, key = lambda x: x[0])
        
        minLastInterval = sorted_intervals[0][0]
        maxLastInterval = sorted_intervals[0][1]
        
        ans = len(sorted_intervals)
        
        for i in range(1,len(sorted_intervals)):

            if sorted_intervals[i][1] <= maxLastInterval:
                ans -= 1
            else:
                if minLastInterval ==  sorted_intervals[i][0]:
                    ans -= 1                
                minLastInterval = sorted_intervals[i][0]
                maxLastInterval = sorted_intervals[i][1]
            
            
        return ans